### DISCUSSION TITLE
Explicit Solver

### DISCUSSION THREAD

**Discussion by Erol-hub** [Computational-Mechanics-Material-Models/chrono-mechanics#7] [Q&A] [answered]
Hi. I tried to use explicit solver as shown below:

auto solver = chrono_types::make_shared<ChSolverMINRES>();
sys.SetSolver(solver);
solver->SetMaxIterations(200);
solver->SetTolerance(1e-12);
solver->EnableDiagonalPreconditioner(true);
solver->EnableWarmStart(true);  
sys.SetSolverForceTolerance(1e-10);
     
sys.SetTimestepperType(ChTimestepper::Type::EULER_EXPLICIT);

But i couldnt manage it. It diverges immediately (free from the time step). Is there any other way to enable explicit solver. Thanks.

**Comment by mtroemner** [comment, marked as answer]
From Alessandro Tasora: 

"Is there a need for the expicit integration? In general the most stable and robust integrator is the default one (Euler implicit linearized). Explicit integration usually requires very short time steps, depending on the material properties that you are using, often 2-3 orders of magnitude smaller than the time step for implicit Euler.

Also: do you have some constraint (ChLinkMate... or ChLinkLock...)  in your system?

Are you using ChSystemSMC? or ChSystemNSC?

Another thing: I see that you enable some non-default solver, as I read  " auto solver = chrono_types::make_shared()"  .. but which type? you did not write the template argument in the text.. was it something like
      auto solver = chrono_types::make_shared<ChSolverMINRES>();
or what? I must say that, if you use some finite element with non-trivial material properties, then MINRES is not the best option, because it may converge slowly, especially if one has very stiff items. A much better option is the PARDISO MKL solver (but you have to install MKL from Intel, and enable the PARDISO MKL module by running CMake again).
Please look at  demo_FEA_contacts_SMC.cpp for an example."
